### ISSUE TITLE
InterfaceId values

### ISSUE THREAD

**Issue by padamjain** [deltics/delphi.libs#10]
Hi,

I am using Bonjour libs in my VCL application.
There is a case when my pc has IPv4 disabled from network adapter and only IPv6 is enabled. 

The above condition is leading to strange value of InterfaceIndex. 
When IPv6 and IPv4 are enabled : 7
When IPv6 is enabled but IPv4 is disabled : 10000007
When IPv6 is disabled and IPv4 is enabled : 7

The correct value is 7.  do you have any idea how to interpret that strange (long) value. The same is response from dns-sd -B command. so its not an issue with component but its causing some problems in my application.

**Comment by deltics** [comment]
That is a curious one and I'm afraid I don't have an answer.  The fact that it is consistent with dns-sd at least tells me that it is some aspect of ZeroConf/Bonjour itself, but I don't have an explanation for it and have been unable to find anything relating to this online (as I'm sure you've also already tried).

I'd be interested to understand what problems it is causing your application though ?

**Comment by padamjain** [comment]
Yes, I am trying to get the reason for this online at several places even on apple developer forum.
The issue is that my application uses this interface index to append in the URL with ipv6 address. By doing this my application reaches so fast. but when the id is wrong , the devices are unreachable.
